docs/build/guides/conventions/upgrading-contracts.mdx

```diff
@@ -0,0 +1,333 @@
+    let client = new_contract::Client::new(&env, &contract_id);
+    assert_eq!(1010101, client.new_v2_fn());
+}
+
+```
+
+We first import Wasm files for both contracts:
+
+```rust
+mod old_contract {
+    soroban_sdk::contractimport!(
+        file =
+            "target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm"
+    );
+}
+
+mod new_contract {
+    soroban_sdk::contractimport!(
+        file = "../new_contract/target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm"
+    );
+}
+```
+
+We register the old contract, intialize it with an admin, and verify the version it returns. The note in the code below is important:
+
+```rust
+// Note that we use register_contract_wasm instead of register_contract
+// because the old contracts Wasm is expected to exist in storage.
+let contract_id = env.register_contract_wasm(None, old_contract::Wasm);
+
+let client = old_contract::Client::new(&env, &contract_id);
+let admin = Address::random(&env);
+client.init(&admin);
+
+assert_eq!(1, client.version());
+```
+
+We install the new contract's Wasm:
+
+```rust
+let new_wasm_hash = install_new_wasm(&env);
+```
+
+Then we run the upgrade, and verify that the upgrade worked:
+
+```rust
+client.upgrade(&new_wasm_hash);
+assert_eq!(2, client.version());
+```
+
+## Build the contract
+
+To build the contract `.wasm` files, run `stellar contract build` in both `upgradeable_contract/old_contract` and `upgradeable_contract/new_contract` in that order.
+
+Both `.wasm` files should be found in both contract `target` directories after building both contracts:
+
+```
+target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm
+```
+
+```
+target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm
+```
+
+## Run the contract
+
+If you have [`stellar-cli`] installed, you can invoke contract functions. Deploy the old contract and install the Wasm for the new contract.
+
+Navigate to to `upgradeable_contract/old_contract`
+
+```sh
+stellar contract deploy \
+  --wasm target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_old_contract.wasm  --source alice  \
+  --network testnet
+```
+
+- When you deploy a smart contract to a network, you need to specify an identity that will be used to sign the transactions. Change the `alice` to your own [identity].
+
+[identity]: ../../smart-contracts/getting-started/setup#configure-an-identity
+
+You should see this similar contract ID after you have ran the deploy command:
+
+```
+CAS6FKBXGVXFGU2SPPPJJOIULJNPMPR6NVKWLOQP24SZJPMB76TGH7Y3
+```
+
+Navigate to `upgradeable_contract/new_contract` and run the following command:
+
+```
+stellar contract install \
+--source-account alice \
+--wasm target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm \
+--rpc-url https://soroban-testnet.stellar.org:443 \--network-passphrase "Test SDF Network ; September 2015" \
+--network testnet
+```
```

Comment:
````suggestion
```bash
stellar contract install \
  --wasm target/wasm32-unknown-unknown/release/soroban_upgradeable_contract_new_contract.wasm \
  --source alice \
  --network testnet
```
````

Since we're specifying `--network testnet`, we don't need to specify the RPC URL or network passphrase parameters (in fact, iirc they'll conflict and the command will fail?). Also, indentation and code language.